src/hooks/useContactPartnershipStatuses.ts

```diff
@@ -0,0 +1,88 @@
+import { useTranslation } from 'react-i18next';
+import { useApiConstants } from 'src/components/Constants/UseApiConstants';
+import { PhaseEnum } from 'src/graphql/types.generated';
+import { getLocalizedContactStatus } from 'src/utils/functions/getLocalizedContactStatus';
+
+export type ContactStatuses = Record<
+  string,
+  {
+    name: string;
+    translated: string;
+    phase: PhaseEnum | null;
+  }
+>;
+
+export type StatusArray = {
+  name: string;
+  translated: string;
+  phase: PhaseEnum | null;
+  id: string;
+}[];
+
+export const useContactPartnershipStatuses = () => {
+  const constants = useApiConstants();
+  const { t } = useTranslation();
+  const phases = constants?.phases;
+  const statuses = constants?.statuses;
+
+  const otherStatuses = {
+    NULL: {
+      name: 'null',
+      translated: t('-- None --'),
+      phase: null,
+    },
+    ACTIVE: {
+      name: 'active',
+      translated: t('-- All Active --'),
+      phase: null,
+    },
+    HIDDEN: {
+      name: 'hidden',
+      translated: t('-- All Hidden --'),
+      phase: null,
+    },
+  };
+
+  const contactStatuses: ContactStatuses = phases
```

Comment:
Four these found status transformations, can you add a comment showing an example of what the object loooks like? This isn't simple logic, so it might be good to add tests for this hook too.